serialise/src/Codec/Serialise/Class.hs

```diff
@@ -204,6 +207,13 @@ instance Serialise a => Serialise (NonEmpty.NonEmpty a) where
 --------------------------------------------------------------------------------
 -- Primitive and integral instances
 
+#if MIN_VERSION_base(4,8,0)
+-- | @since 0.2.3.1
+instance Serialise Void where
```

Comment:
Depend conditionally on `void` package.

Reply:
I don't know how to make sense of that, can you clarify? There are no unconditional dependencies in the cabal file at the moment, and this conditional definition is similar to the other ones in the same file - i.e. it doesn't exist if the version of base being used doesn't support it. By contrast the `void` package seems to be designed as an unconditional dependency that shims in support for older `base`, but I don't know if the maintainers want to do that.

Reply:
https://oleg.fi/gists/posts/2019-06-03-compat-packages.html#s:2

Reply:
OK that makes sense now, so I would do something like if impl(ghc < 7.10) build-depends: void` in the cabal file and drop the `MIN_VERSION_base(4,8,0)` check for `Data.Void` in `Serialise.Class`.

Could one of the cborg maintainers confirm that you want this? I note the current approach of this package doesn't do this for e.g. `Data.Semigroup` - there is no attempt to shim in `semigroups` for older base.

Reply:
I'd say the `semigroups` should also be shimmed, `serialise` depends on `semigroups` transitively anyways.